spring-batch-core/src/main/java/org/springframework/batch/core/step/tasklet/TaskletStep.java

```diff
@@ -175,8 +174,8 @@ public void registerChunkListener(ChunkListener listener) {
 	 * @param listeners an array of listener objects of known types.
 	 */
 	public void setChunkListeners(ChunkListener[] listeners) {
-		for (int i = 0; i < listeners.length; i++) {
-			registerChunkListener(listeners[i]);
+		for (ChunkListener listener : listeners) {
+			registerChunkListener(listener);
```

Comment:
This is not related to the removal of the deprecated `TransactionSynchronizationAdapter`. I prefer to not include unrelated changes in the same commit. I can remove it on merge. You can open a separate PR for this refactoring if you want.

Reply:
Reverted the enhanced for loops.